ambry-store/src/test/java/com.github.ambry.store/BlobStoreStatsTest.java

```diff
@@ -241,8 +315,351 @@ public void testValidDataSizeAfterDeletes() throws InterruptedException, StoreEx
+      verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(i, 0L));
+    }
+    // advance time to let the added puts to expire
+    long timeToLiveInMs = expiresAtInMs - state.time.milliseconds() < 0 ? 0 : expiresAtInMs - state.time.milliseconds();
+    state.advanceTime(timeToLiveInMs + Time.MsPerSec);
+    for (long i = 0; i <= state.time.milliseconds() + TEST_TIME_INTERVAL_IN_MS; i += TEST_TIME_INTERVAL_IN_MS) {
+      verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(i, 0L));
+    }
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    assertEquals("Throttle count shouldn't have changed", throttleCountBeforeRequests,
+        mockThrottler.throttleCount.get());
+    blobStoreStats.close();
+  }
+
+  /**
+   * Test to verify new entries after a scan are properly counted to keep the current {@link ScanResults} relevant
+   * before the next scan. In addition, verify that requests within the coverage are served without triggering any scan.
+   * @throws StoreException
+   * @throws InterruptedException
+   * @throws IOException
+   */
+  @Test
+  public void testBucketingWithNewEntriesAfterScan() throws StoreException, InterruptedException, IOException {
+    assumeTrue(bucketingEnabled);
+    CountDownLatch scanStartedLatch = new CountDownLatch(1);
+    CountDownLatch queueProcessedLatch = new CountDownLatch(1);
+    MockThrottler mockThrottler = new MockThrottler(scanStartedLatch, new CountDownLatch(0));
+    throttlers.put(BlobStoreStats.IO_SCHEDULER_JOB_TYPE, mockThrottler);
+    BlobStoreStats blobStoreStats =
+        setupBlobStoreStats(2 * (int) (state.time.milliseconds() / BUCKET_SPAN_IN_MS), state.time.milliseconds());
+    // proceed only when the scan is started
+    assertTrue("IndexScanner took too long to start", scanStartedLatch.await(5, TimeUnit.SECONDS));
+    advanceTimeToNextSecond();
+    verifyAndGetLogSegmentValidSize(blobStoreStats, new TimeRange(state.time.milliseconds(), 0L));
+    verifyAndGetContainerValidSize(blobStoreStats, state.time.milliseconds());
+    int throttleCountBeforeRequests = mockThrottler.throttleCount.get();
+    // add 3 put with expiry
+    long expiresAtInMs = state.time.milliseconds() + TimeUnit.SECONDS.toMillis(30);
+    List<IndexEntry> newPutEntries = state.addPutEntries(3, CuratedLogIndexState.PUT_RECORD_SIZE, expiresAtInMs);
+    for (IndexEntry entry : newPutEntries) {
+      blobStoreStats.handleNewPutEntry(entry.getValue());
+    }
+    // delete the first new put with expiry
+    MockId firstPutWithExpiry = getIdToDelete(newPutEntries.get(0).getKey());
+    newDelete(blobStoreStats, firstPutWithExpiry);
+    // a probe put with a latch to inform us about the state of the queue
+    blobStoreStats.handleNewPutEntry(new MockIndexValue(queueProcessedLatch, state.index.getCurrentEndOffset()));
+    assertTrue("QueueProcessor took too long to process the new entries",
+        queueProcessedLatch.await(5, TimeUnit.SECONDS));
```

Comment:
why wait for 5 secs. We know queueProcessor should kick in for every 1 sec right. You can give 2 more secs for processing. 5 secs is too much.